src/Slider/Slider.jsx

```diff
@@ -521,6 +537,39 @@ const Slider = class Slider extends React.Component {
     return null;
   }
 
+  checkInfinite() {
+    let skipToSlide = null;
+
+    const {
+      carouselStore,
+      currentSlide,
+      infinite,
+      totalSlides,
+    } = this.props;
+
+    if (infinite) {
+      if (currentSlide > totalSlides) {
+        skipToSlide = 1;
+      } else if (currentSlide < 0) {
+        skipToSlide = totalSlides - 1
+      }
+
+      if (skipToSlide) {
+        this.setState({
+          disableAnimationTemp: true
+        });
+        carouselStore.setStoreState({
+          currentSlide: skipToSlide
+        });
+        setTimeout(() => {
```

Comment:
Can you provide more detail on this here? We are disabling animation, setting the store, and re-enabling it with setTImeout?